integration-tests/apisonator.go

```diff
@@ -376,6 +377,8 @@ func executeHTTPRequest(method string, url string, data *[]byte) (*http.Response
 func (backend *Backend) Push(stateName string, params []interface{}) error {
 	defer func() {
```

Comment:
There is an issue with the use of `defer` here. Defer executes after the surrounding function returns. So here if you do this way, even when the functions in the switch case returns an error , you are adding a new state to the backend states. I fixed this issue in my PR and should be file after you do a rebase.

Reply:
I was thinking of using named return value to tackle this problem but your solution is also good.